### DISCUSSION TITLE
qtile-setup

### DISCUSSION THREAD

**Discussion by zordsdavini** [qtile/qtile#2560] [Ideas]
It would be good for new users to have scenario to configure Qtile by selecting answers. It should generate (or update) config. 
Even each new release I would like to go through the questionnaire and find what is new.
From architecture side I'd suggest to have reusable config object. (Maybe it could be reused for GUI configuring)

**Comment by zordsdavini** [comment]
Do we have method to get config and after write back?

**Comment by Roger** [comment]
i think it should be kinda hard to update the config, since it's python code

**Comment by zordsdavini** [comment]
Maybe we shoul try not to update, bu crear new one. It could be initial
stage
2015 Lapkr. 24 17:20 "Roger" notifications@github.com rašė:

**Comment by Roger** [comment]
would you mind making an example of a qtile-setup session? i can't think in a lot of things that could be asked to build a new config

**Comment by zordsdavini** [comment]
Okey. I'll try to make some example
2015 Lapkr. 24 18:40 "Roger" notifications@github.com rašė:

**Comment by Infinemen** [comment]
I follow the guide strictly, but qtile still cannot start. Here is the messages:
libqtile **init**.py:safe_import():L47 can't import widget: '.launchBar.Launchbar', No module namd 'xdg'
libqtile **init**.py:safe_import():L47 can't import widget: '.mpriswidget.Mpris, No module namd 'dbus'
libqtile **init**.py:safe_import():L47 can't import widget: '.mpdwidget.Mpd', No module namd 'mpd'
balbala
libqtile qtile.py:make_qtile():L119 qtile crash during startup
Traceback
File "/usr/local/lib/python3.4/dist-packages/libtile/scripts/qtile.py" ,line116 in make_qtile stste=options.state
File "/usr/local/lib/python3.4/dist-packages/libtile/manager.py" raise QtileErrors:('No DISPLAY set.
')
libqtile.utils.QtileErrors:No DISPLAY set.
Waiting for helps
Thanks!

**Comment by flacjacket** [comment]
@Infinemen How are you starting qtile? You can't just run it from the command line, it needs to be run through an X11 session (via `startx` or with your display manager(e.g. gdm, kdm)), those will set the `DISPLAY` environment variable, which is needed by qtile to know what display it is on.

**Comment by flacjacket** [comment]
Also, you probably shouldn't hijack this issue, you can either open a new issue, or for support things like this, it is probably better to email the google group list, qtile-dev@googlegroups.com

**Comment by Infinemen** [comment]
@flacjacket Thank you so much. i am unfamiliar with github. And i am Chinese. I can't find any solutions written in Chinese. But i like the style of qtile. I don't want to give up halfway. So i make a try here. Cause i saw the last comment was in 2015, i thought i would not get any answers. But i do (or did?? forgive my poor English) ! You helped me a lot. Not just this question. You give me hope to let me believe that i can make it.  Thanks!

**Comment by Infinemen** [comment]
@flacjacket And yes i ran it from command line. I saw the googlegroups in the homepage. But you know, it's not easy for Chinese to view google. I will try start qtile through display manager. If i fail again, i will ask in the googlegroups.

**Comment by flacjacket** [comment]
@Infinemen no problem, feel free to open a new issue on github if you are having other problems: https://github.com/qtile/qtile/issues/new

**Comment by andering** [comment]
I think i had same issue, just can't recall what was the reason,
if you send me your config.py i can look at it.

**Comment by Infinemen** [comment]
@andering  I have already solved the problem, but still thank you!

**Comment by Infinemen** [comment]
@flacjacket OK, i will.Your advice helps me out, thank you so much!

**Comment by cblegare** [comment]
Back to the original subject:

This could be achieved simply enough for the initial configuration using a [cookiecutter template](https://github.com/audreyr/cookiecutter) or something similar.

**Comment by zordsdavini** [comment]
Nice. Will check that.
